src/pages/commits/TupleSelectWithRegexConditional/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import {
+  SegmentedControl,
+  SegmentedControlOption,
+} from "@leafygreen-ui/segmented-control";
+import TupleSelect from "components/TupleSelect";
+import { size } from "constants/tokens";
+
+interface TupleSelectWithRegexConditionalProps
+  extends React.ComponentProps<typeof TupleSelect> {}
+
+/**
+ * TupleSelectWithRegexConditional is a wrapper around TupleSelect that allows the user to toggle between regex and exact match
+ * @param param0
+ * @returns
+ */
+const TupleSelectWithRegexConditional: React.VFC<
+  TupleSelectWithRegexConditionalProps
+> = ({ onSubmit, validator, ...rest }) => {
+  const [type, setType] = useState("regex");
+  const isRegex = type === "regex";
+
+  const handleOnSubmit = ({
+    category,
+    value,
+  }: {
+    category: string;
+    value: string;
+  }) => {
+    onSubmit({ category, value: isRegex ? value : escapeRegex(value) });
+  };
+
+  return (
+    <TupleSelect
+      {...rest}
+      onSubmit={handleOnSubmit}
+      aria-label="tuple-select-with-regex"
+      validator={isRegex ? validator : () => true}
+      label={
+        <>
+          Add New Filter
+          <PaddedSegmentedControl
+            size="small"
+            onChange={setType}
+            value={type}
+            aria-controls="tuple-select-with-regex"
+          >
+            <SegmentedControlOption value="regex" data-cy="tuple-select-regex">
+              REGEX
+            </SegmentedControlOption>
+            <SegmentedControlOption value="exact" data-cy="tuple-select-exact">
+              EXACT
+            </SegmentedControlOption>
+          </PaddedSegmentedControl>
+        </>
+      }
+    />
+  );
+};
+
+const PaddedSegmentedControl = styled(SegmentedControl)`
+  margin-left: ${size.xs};
+`;
+
+const escapeRegex = (str: string) =>
```

Comment:
this might be a separate ticket, but I noticed that the params don't get properly escaped in the URL

Reply:
How so? they seem to be escaped to me.
<img width="589" alt="image" src="https://user-images.githubusercontent.com/4605522/215116191-2616879f-1010-4fdb-af7c-b160826319f2.png">

Reply:
Oh sorry, this was a quirk in Firefox I never noticed where the URL isn't escaped but when you copy it it is